### PR TITLE
Create edit in positions

### PR DESCRIPTION
**Pull request by ANGIECATHERINEP**
Create edit in positions

### CODE REVIEW

**Review by luisalejandrosalazar — APPROVED** [review]
Meets the requirements.

File: src/pages/privileges/outlets/forms/InitializerForm/index.tsx [hunk 1]
```diff
@@ -7,22 +7,36 @@ import {
...
   withSubmitButtons?: boolean;
+  id?: string;
+  property?: string;
+  keyData?: string;
+  nameDB?: string;
+  propertyData?: string;
+  handleSubmit: (renderForm: IAssignmentFormEntry[]) => void;
   onHasChanges?: (hasChanges: boolean) => void;
-  readOnly?: boolean;
+  editItemData?: (props: functionById) => Promise<unknown>;
```
[the remark this answers is not in the document]
Comment (ANGIECATHERINEP) [reply src/pages/privileges/outlets/forms/InitializerForm/index.tsx]: What happens is that the functionById interface such as the edit and delete functions are located in src\mocks\utils\dataMock.service.ts

File: src/pages/privileges/outlets/positions/add-position/forms/GeneralInformationForm/index.tsx [hunk 2]
```diff
@@ -1,57 +1,147 @@
...
+          setMessage({
+            visible: true,
+            data: generalMessage.failed,
+          });
+        }
+        formik.handleSubmit();
       });
     };
+
+    const disabledButtons = (valueCompare: IGeneralInformationEntry) =>
```
[the remark this answers is not in the document]
Comment (ANGIECATHERINEP) [reply src/pages/privileges/outlets/positions/add-position/forms/GeneralInformationForm/index.tsx]: Yes sir, the form works correctly

File: src/pages/privileges/outlets/positions/add-position/forms/GeneralInformationForm/interface.tsx [hunk 3]
```diff
@@ -1,67 +1,119 @@
...
+    message,
+    disabledButtons,
+    handleCloseSectionMessage,
+    handleReset,
+    handleChangeForm,
+    handleSubmitForm,
+  } = props;
+
+  if (withSubmitButtons) {
+    return (
```
[the remark this answers is not in the document]
Comment (ANGIECATHERINEP) [reply src/pages/privileges/outlets/positions/add-position/forms/GeneralInformationForm/interface.tsx]: resolved

File: src/pages/privileges/outlets/forms/InitializerForm/index.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (carin91) [reply src/pages/privileges/outlets/forms/InitializerForm/index.tsx]: Pending

File: src/pages/privileges/outlets/positions/add-position/forms/GeneralInformationForm/interface.tsx [hunk 4]
```diff
@@ -1,67 +1,119 @@
...
+          <RenderMessage
+            message={message}
+            handleCloseMessage={handleCloseSectionMessage}
+            onMessageClosed={handleReset}
+          />
+        )}
+      </>
+    );
+  }
+  return <>{RenderFormFields(formik, handleChangeForm)}</>;
```
[the remark this answers is not in the document]
Comment (carin91) [reply src/pages/privileges/outlets/positions/add-position/forms/GeneralInformationForm/interface.tsx]: Pending.

You are passing a react component as a function, therefore, you are not giving the chance to react to calculate which component to rerender or not.

read:
https://react.dev/reference/rules/react-calls-components-and-hooks

**Review by carin91 — CHANGES_REQUESTED** [review]
Verify that all of the edited files are without warnings / errors.

image

**Review by ANGIECATHERINEP — COMMENTED** [review]
additionally adjust the file with errors that you mentioned, but this file is not found in positions

**Review by carin91 — APPROVED** [review]

**Pull request merged**